src/material/datepicker/date-range-input-parts.ts

```diff
@@ -75,8 +77,13 @@ class MatDateRangeInputPartBase<D> extends MatDatepickerInputBase<D> implements
     @Optional() public _parentForm: NgForm,
     @Optional() public _parentFormGroup: FormGroupDirective,
     @Optional() dateAdapter: DateAdapter<D>,
-    @Optional() @Inject(MAT_DATE_FORMATS) dateFormats: MatDateFormats) {
+    @Optional() @Inject(MAT_DATE_FORMATS) dateFormats: MatDateFormats,
+
+    // TODO(crisbeto): this will be provided by the datepicker eventually.
+    // We provide it here for the moment so we have something to test against.
+    model: MatDateSelectionModel<DateRange<D | null>>) {
```

Comment:
the type seems wrong, shouldn't it be `MatDateSelectionModel<D, DateRange<D | null>>`?

Reply:
Hmm, you're right, this seems like a code smell. I think for it to actually be typed correctly the base class would have to be something like:

```
MatDatepickerInputBase<D, S, M extends MatDateSelectionModel<D, S>>
```

This also seems a little excessive.